impeller/geometry/path.cc

```diff
@@ -299,33 +299,37 @@ std::optional<std::pair<Point, Point>> Path::GetMinMaxCoveragePoints() const {
 
   std::optional<Point> min, max;
 
-  auto clamp = [&min, &max](const std::vector<Point>& extrema) {
-    for (const auto& extremum : extrema) {
-      if (!min.has_value()) {
-        min = extremum;
-      }
-
-      if (!max.has_value()) {
-        max = extremum;
-      }
+  auto clamp = [&min, &max](const Point& point) {
+    if (min.has_value()) {
+      min->x = std::min(min->x, point.x);
+      min->y = std::min(min->y, point.y);
+    } else {
+      min = point;
+    }
 
-      min->x = std::min(min->x, extremum.x);
-      min->y = std::min(min->y, extremum.y);
-      max->x = std::max(max->x, extremum.x);
-      max->y = std::max(max->y, extremum.y);
+    if (max.has_value()) {
+      max->x = std::max(max->x, point.x);
+      max->y = std::max(max->y, point.y);
+    } else {
+      max = point;
     }
   };
 
   for (const auto& linear : linears_) {
-    clamp(linear.Extrema());
+    clamp(linear.p1);
+    clamp(linear.p2);
   }
 
   for (const auto& quad : quads_) {
-    clamp(quad.Extrema());
+    for (const Point& point : quad.Extrema()) {
```

Comment:
I didn't think this would be all that expensive. But finding the extrema for all components is overkill. Since the curve would lie entirely within the convex hull of its control points, perhaps we should just cycle over the control points instead? This would make coverage calculation way faster for not a lot of wastage.

cc @bdero

Reply:
I'm a bit worried that the degenerate case is a bit too extreme for min/maxing the control points and we're going to get some bad(at least for cubics):

![Screen Shot 2022-11-21 at 11 00 34 PM](https://user-images.githubusercontent.com/919017/203246862-de16e94f-334f-4f17-bf73-65fb6643733d.png)
Green is coverage as it's currently computed.

I took a look over this code and noticed other easy improvements we can make: For cubics, we find coverage by taking the analytical derivative (a quadratic curve) and then finding the up-to-2 solutions for y=0 for each dimension -- we can improve things here by keeping more stuff on the stack, but so far so good.
But for quadratics, `QuadraticPathComponent::Extrema()` "upgrades" the curve to quadratic form and then finds the quadratic extrema. Instead, we should just take the derivative of the quadratic (a line) and include the y=0 solution for each direction.

Reply:
We should file an issue for that. In the meantime, this patch is good to go.